app/src/androidTest/java/com/github/sdpteam15/polyevents/MyLocationTests.kt

```diff
@@ -0,0 +1,155 @@
+        val bottomNavigationItemView = onView(
+            Matchers.allOf(
+                withId(R.id.ic_map), ViewMatchers.withContentDescription("Map"),
+                childAtPosition(
+                    childAtPosition(
+                        withId(R.id.navigation_bar),
+                        0
+                    ),
+                    1
+                ),
+                ViewMatchers.isDisplayed()
+            )
+        )
+        bottomNavigationItemView.perform(click())
+
+        val imageButton = onView(
+            Matchers.allOf(
+                withId(R.id.id_location_button),
+                ViewMatchers.withParent(ViewMatchers.withParent(withId(R.id.fl_wrapper))),
+                ViewMatchers.isDisplayed()
+            )
+        )
+        imageButton.check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+
+        val imageButton2 = onView(
+            Matchers.allOf(
+                withId(R.id.id_locate_me_button),
+                ViewMatchers.withParent(ViewMatchers.withParent(withId(R.id.fl_wrapper))),
+                ViewMatchers.isDisplayed()
+            )
+        )
+        imageButton2.check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+    }
+
+    @Test
+    fun denyPermissionKeepsLocationOff() {
+        // Revoke all the permissions
+        //InstrumentationRegistry.getInstrumentation().uiAutomation.executeShellCommand("pm reset-permissions")
+
+        val bottomNavigationItemView = onView(
+            Matchers.allOf(
+                withId(R.id.ic_map), ViewMatchers.withContentDescription("Map"),
+                childAtPosition(
+                    childAtPosition(
+                        withId(R.id.navigation_bar),
+                        0
+                    ),
+                    1
+                ),
+                ViewMatchers.isDisplayed()
+            )
+        )
+        bottomNavigationItemView.perform(click())
+
+        denyPermissions()
+
+        // Click on the "location" button to try to activate it.
+        onView(withId(R.id.id_location_button)).perform(click())
+
+        // Check the location is not enabled
+        onView(withTagValue(equalTo(R.drawable.ic_location_off)))
+    }
+
+    private fun childAtPosition(
+        parentMatcher: Matcher<View>, position: Int
+    ): Matcher<View> {
+
+        return object : TypeSafeMatcher<View>() {
+            override fun describeTo(description: Description) {
+                description.appendText("Child at position $position in parent ")
+                parentMatcher.describeTo(description)
+            }
+
+            public override fun matchesSafely(view: View): Boolean {
+                val parent = view.parent
+                return parent is ViewGroup && parentMatcher.matches(parent)
+                        && view == parent.getChildAt(position)
+            }
+        }
+    }
+
+    private fun denyPermissions() {
+        if (Build.VERSION.SDK_INT >= 23) {
```

Comment:
Check not needed anymore, we've upgraded the min sdk version to 26 :)

Reply:
Oups, you're right. This was something I took from somewhere else and did not check, thanks :)

Reply:
And actually there is a bug in this part, I'm looking into it.